src/main/java/net/mcreator/preferences/PreferencesManager.java

```diff
@@ -20,42 +21,204 @@
 
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.internal.LinkedTreeMap;
 import net.mcreator.io.FileIO;
 import net.mcreator.io.UserFolderManager;
+import net.mcreator.plugin.events.ApplicationLoadedEvent;
+import net.mcreator.preferences.data.Preferences;
+import net.mcreator.preferences.entries.PreferenceEntry;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.awt.*;
 import java.io.File;
+import java.util.List;
+import java.util.*;
 
 public class PreferencesManager {
 
 	private static final Logger LOG = LogManager.getLogger("Preferences Manager");
 
-	public static PreferencesData PREFERENCES = new PreferencesData();
+	private static final File file = UserFolderManager.getFileFromUserFolder("preferences.json");
 
 	private static final Gson gson = new GsonBuilder().setPrettyPrinting().setLenient().create();
-	private static final File preferencesFile = UserFolderManager.getFileFromUserFolder("preferences");
 
-	public static void loadPreferences() {
-		if (!UserFolderManager.getFileFromUserFolder("preferences").isFile()) {
-			storePreferences(new PreferencesData());
-			LOG.info("Preferences not created yet. Loading defaults.");
+	/**
+	 * <p>Store values when the preferences' file is loaded</p>
+	 */
+	private static Map<String, Map<String, List<LinkedTreeMap<?, ?>>>> loadedPreferences;
+
+	/**
+	 * <p>Store all preferences using an identifier </p>
+	 */
+	private static Map<String, List<PreferenceEntry<?>>> PREFERENCES_REGISTRY;
+
+	/**
+	 * <p>Default values for all preferences. Used when resetting preferences.</p>
+	 */
+	private static Map<String, List<PreferenceEntry<?>>> DEFAULT_PREFERENCES;

```

Comment:
I think instead of this Map, each PreferenceEntry should have final defaultValue field initialized in constructor from value and this should be used for default value